### ISSUE TITLE
Changes to an excluded file is triggering a restart

### ISSUE THREAD

**Issue by kichalla** [aspnet/dotnet-watch#90]
Repro: https://github.com/kichalla/dotnetwatchtestapp/
Git tag: excludedfile_causing_restart

When I change the following excluded Program.cs file, a restart happens
```
  "buildOptions": {
    "emitEntryPoint": true,
    "compile": {
      "excludeFiles": [
        "temp/Program.cs"
      ]
    }
  }
```

Expected: restart should not happen and the excluded files should be ignored from being watched
Actual: restart is triggered

**Comment by victorhurdugaci** [comment]
Interesting, I wonder if it's caused by the recent project.json chances. There's a test already for that and it passes but it uses the old project format:

https://github.com/aspnet/dotnet-watch/blob/dev/test/TestApps/GlobbingApp/project.json#L11

**Comment by muratg** [comment]
@victorhurdugaci Let's investigate. If it's not a super simple fix, we can move to 1.0.1

**Comment by victorhurdugaci** [comment]
Just as I suspected, this was caused by the new project structure/reader
